src/Interpreters/InterpreterSelectQuery.cpp

```diff
@@ -1249,7 +1250,12 @@ void InterpreterSelectQuery::executeImpl(QueryPlan & query_plan, const BlockInpu
             // 4) preliminary distinct.
             // Some of these were already executed at the shards (first_stage),
             // see the counterpart code and comments there.
-            if (expressions.need_aggregate)
+            if (from_aggregation_stage)
+            {
+                if (query_analyzer->hasWindow())
+                    throw Exception("Window functions does not support processing from WithMergeableStateAfterAggregation", ErrorCodes::NOT_IMPLEMENTED);
```

Comment:
Should probably work for distributed_group_by_no_merge = 1,

By the way, the desription says: ` If 2 - same as 1 but also apply ORDER BY and LIMIT stages` -- how does it merge the ordered block on the coordinator? The resulting order is ultimately unspecified?

Reply:
>Should probably work for distributed_group_by_no_merge = 1,

Yes, and it will work, `from_aggregation_stage` is set only for `distributed_group_by_no_merge=2`, since `distributed_group_by_no_merge=1` is simply proxying the data

>By the way, the desription says: If 2 - same as 1 but also apply ORDER BY and LIMIT stages -- how does it merge the ordered block on the coordinator? The resulting order is ultimately unspecified?

- 1 -- each shard process query up to the Complete stage, initiator just proxying the data
- 2 -- each shard process query up to the WithMergeableStateAfterAggregation, initiator applies ORDER BY (merge) and LIMIT

Reply:
Wait, this is the opposite of how I read it. Should it say `If 2 - same as 1 but also apply ORDER BY and LIMIT stages _on initiator_`? Then we can probably enable window functions for 2 but not for 1.

Reply:
>Should it say If 2 - same as 1 but also apply ORDER BY and LIMIT stages _on initiator_? 

Agree, let's update the description.

>Then we can probably enable window functions for 2 but not for 1.

It works for `distributed_group_by_no_merge=1` because initiator does do anything (and actually anyway who uses it should not expect that it will work, but I guess an error can be thrown for queries with window functions in this case), so yes it works incorrectly.

It does not works for `distributed_group_by_no_merge=2` because it tries to execute function one more time, but shards does not sends the columns that is required for those functions.
I guess it may work, but I though about fixing this separately, since right now you can't use `distributed_group_by_no_merge=2` even w/o window functions (with simple `func(aggregate function)` queries).